tracking.el

```diff
@@ -390,12 +399,23 @@ This is usually called via `window-configuration-changed-hook'."
         (tracking-remove-buffer buffer))))))
 
 ;;; Helper functions
+(defun tracking-filter-mode (buffer)
+  "If true, BUFFER should be filtered out of shortened buffers."
+  (unless (eq tracking-shorten-modes t)
+    (not (memq
+          (buffer-local-value 'major-mode buffer)
+          tracking-shorten-modes))))
+
 (defun tracking-shorten (buffers)
   "Shorten BUFFERS according to `tracking-shorten-buffer-names-p'."
   (if tracking-shorten-buffer-names-p
-      (let ((all (shorten-strings (mapcar #'buffer-name (buffer-list)))))
+      (let ((all (shorten-strings
+                  (mapcar #'buffer-name
+                          (cl-remove-if #'tracking-filter-mode
+                                        (buffer-list))))))
         (mapcar (lambda (buffer)
-                  (let ((short (cdr (assoc buffer all))))
+                  (let ((short (or (cdr-safe (assoc buffer all))
+                                   buffer)))
```

Comment:
That doesn't make sense. You cannot use `length` on a buffer and you cannot use buffers/string interchangeably when it comes to text properties.

Reply:
I'm still not 100% sure, but I think in this case, the buffer variable must be a string. Above this assoc is used to index into an alist from buffer string -> shortened string (the conversion is done by buffer-name above), so I think that if it was a buffer that would break first (?)

Reply:
Fair enough.